pkg/dist/extension_descriptor.go

```diff
@@ -1,22 +1,76 @@
 package dist
 
 import (
+	"fmt"
 	"strings"
 
 	"github.com/buildpacks/lifecycle/api"
+
+	"github.com/buildpacks/pack/internal/style"
 )
 
 type ExtensionDescriptor struct {
-	WithAPI  *api.Version `toml:"api"`
-	WithInfo ModuleInfo   `toml:"extension"`
+	WithAPI          *api.Version `toml:"api"`
+	WithInfo         ModuleInfo   `toml:"extension"`
+	WithTargets      []Target     `toml:"targets,omitempty"`
+	WithWindowsBuild bool
+	WithLinuxBuild   bool
 }
 
 func (e *ExtensionDescriptor) EnsureStackSupport(_ string, _ []string, _ bool) error {
 	return nil
 }
 
-func (e *ExtensionDescriptor) EnsureTargetSupport(_, _, _, _ string) error {
-	return nil
+func (e *ExtensionDescriptor) EnsureTargetSupport(os, arch, distroName, distroVersion string) error {
```

Comment:
@jjbustamante are you aware of any existing logic we can re-use to validate that extensions support targets?

Reply:
I think this code is similar to this [one](https://github.com/buildpacks/pack/blob/main/pkg/dist/buildpack_descriptor.go#L57) maybe we can try to refactor it and externalize it to be reuse in both places?

Reply:
@jjbustamante could you please share the directory where we need to keep the share code !! Any additional information about the specific files or workflow would be super helpful!!

Reply:
@jjbustamante @natalieparellano I think a direct refactoring using a common function might not be feasible due to the structural differences (ExtensionDescriptor and BuildpackDescriptor). While the two structs share similarities, they do have some distinct fields.

Reply:
@natalieparellano @jjbustamante pls review the changes !!